lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt

```diff
@@ -136,20 +236,32 @@ data class TypedData private constructor(
             val type = toVisit.removeFirst()
             val params = types[type] ?: emptyList()
 
-            for (param in params) {
-                val typeStripped = stripPointer(param.type)
-
-                if (types.containsKey(typeStripped) && !deps.contains(typeStripped)) {
-                    deps.add(typeStripped)
-                    toVisit.add(typeStripped)
+            params.forEach { param ->
+                val extractedTypes = when {
+                    param is EnumType -> {
+                        require(revision == Revision.V1) { "'enum' basic type is not supported in revision ${revision.value}." }
+                        listOf(param.contains)
+                    }
+                    param.type.isEnum() -> {
+                        require(revision == Revision.V1) { "Enum types are not supported in revision ${revision.value}." }
+                        extractEnumTypes(param.type)
+                    }
```

Comment:
Should we have these just in case, or would it be enough for us to check `verifyTypes()`

Reply:
I'd leave them